src/main/java/dev/slimevr/vr/processor/skeleton/LegTweakBuffer.java

```diff
@@ -0,0 +1,660 @@
+
+	// the point at which the scalar is at it max or min in a single locked foot
+	// situation
+	private static final float MIN_SCALAR_ACTIVE = 3.0f;
+	private static final float MAX_SCALAR_ACTIVE = 0.2f;
+
+	private float leftFootSensitivityVel = 1.0f;
+	private float rightFootSensitivityVel = 1.0f;
+	private float leftFootSensitivityAccel = 1.0f;
+	private float rightFootSensitivityAccel = 1.0f;
+
+
+	private static final float SKATING_CUTOFF_ENGAGE = SKATING_DISTANCE_CUTOFF
+		* SKATING_LOCK_ENGAGE_PERCENT;
+	private static final float SKATING_VELOCITY_CUTOFF_ENGAGE = SKATING_VELOCITY_THRESHOLD
+		* SKATING_LOCK_ENGAGE_PERCENT;
+	private static final float SKATING_ACCELERATION_CUTOFF_ENGAGE = SKATING_ACCELERATION_THRESHOLD
+		* SKATING_LOCK_ENGAGE_PERCENT;
+	private static final float SKATING_ROTATIONAL_VELOCITY_CUTOFF_ENGAGE = SKATING_ROTVELOCITY_THRESHOLD
+		* SKATING_LOCK_ENGAGE_PERCENT;
+
+	// getters and setters
+	public Vector3f getLeftFootPosition(Vector3f vec) {
+		if (vec == null)
+			vec = new Vector3f();
+
+		return vec.set(leftFootPosition);
+	}
+
+	public void setLeftFootPosition(Vector3f leftFootPosition) {
+		this.leftFootPosition = leftFootPosition.clone();
+	}
+
+	public Vector3f getRightFootPosition(Vector3f vec) {
+		if (vec == null)
+			vec = new Vector3f();
+
+		return vec.set(rightFootPosition);
+	}
+
+	public void setRightFootPosition(Vector3f rightFootPosition) {
+		this.rightFootPosition = rightFootPosition.clone();
+	}
+
+	public Vector3f getLeftKneePosition(Vector3f vec) {
+		if (vec == null)
+			vec = new Vector3f();
+
+		return vec.set(leftKneePosition);
+	}
+
+	public void setLeftKneePosition(Vector3f leftKneePosition) {
+		this.leftKneePosition = leftKneePosition.clone();
+	}
+
+	public Vector3f getRightKneePosition(Vector3f vec) {
+		if (vec == null)
+
+			vec = new Vector3f();
+		return vec.set(rightKneePosition);
+	}
+
+	public void setRightKneePosition(Vector3f rightKneePosition) {
+		this.rightKneePosition = rightKneePosition.clone();
+	}
+
+	public Vector3f getWaistPosition(Vector3f vec) {
+		if (vec == null)
+			vec = new Vector3f();
+
+		return vec.set(waistPosition);
+	}
+
+	public void setWaistPosition(Vector3f waistPosition) {
+		this.waistPosition = waistPosition.clone();
+	}
+
+	public Quaternion getLeftFootRotation(Quaternion quat) {
+		if (quat == null)
+			quat = new Quaternion();
+
+		return quat.set(leftFootRotation);
+	}
+
+	public void setLeftFootRotation(Quaternion leftFootRotation) {
+		this.leftFootRotation = leftFootRotation.clone();
+	}
+
+	public Quaternion getRightFootRotation(Quaternion quat) {
+		if (quat == null)
+			quat = new Quaternion();
+
+		return quat.set(rightFootRotation);
+	}
+
+	public void setRightFootRotation(Quaternion rightFootRotation) {
+		this.rightFootRotation = rightFootRotation.clone();
+	}
+
+	public Vector3f getLeftFootPositionCorrected(Vector3f vec) {
+		if (vec == null)
+			vec = new Vector3f();
+		return vec.set(leftFootPositionCorrected);
```

Comment:
```suggestion

		return vec.set(leftFootPositionCorrected);
```

Reply:
I think I figured that out I was merging in an odd way hopefully it doesn't happen anymore